### PR TITLE
Updated trace_context.proto and associated files.

### PR DESCRIPTION
**Pull request by Vizerai**
trace_context.proto has been changed to include span_options instead of just an is_sampled bool.  Also, trace_id is no longer a struct and just 2 fixed64 values now (trace_id_hi and trace_id_lo).

### CODE REVIEW

**Comment by Vizerai** [comment]
Bazel full build is flaky.  These 8 tests are failing:

//test/core/bad_ssl:bad_ssl_alpn_server                                  FAILED in 0.2s
  /root/.cache/bazel/_bazel_root/d2dc70c3d9da3fab488ba0dcbbd35051/execroot/grpc/bazel-out/local-fastbuild/testlogs/test/core/bad_ssl/bad_ssl_alpn_server/test.log
//test/core/bad_ssl:bad_ssl_cert_server                                  FAILED in 0.2s
  /root/.cache/bazel/_bazel_root/d2dc70c3d9da3fab488ba0dcbbd35051/execroot/grpc/bazel-out/local-fastbuild/testlogs/test/core/bad_ssl/bad_ssl_cert_server/test.log
//test/core/end2end:h2_fd_test@ping                                      FAILED in 0.1s
  /root/.cache/bazel/_bazel_root/d2dc70c3d9da3fab488ba0dcbbd35051/execroot/grpc/bazel-out/local-fastbuild/testlogs/test/core/end2end/h2_fd_test@ping/test.log
//test/core/end2end:h2_proxy_test@compressed_payload                     FAILED in 0.1s
  /root/.cache/bazel/_bazel_root/d2dc70c3d9da3fab488ba0dcbbd35051/execroot/grpc/bazel-out/local-fastbuild/testlogs/test/core/end2end/h2_proxy_test@compressed_payload/test.log
//test/core/end2end:h2_sockpair_1byte_test@ping                          FAILED in 0.1s
  /root/.cache/bazel/_bazel_root/d2dc70c3d9da3fab488ba0dcbbd35051/execroot/grpc/bazel-out/local-fastbuild/testlogs/test/core/end2end/h2_sockpair_1byte_test@ping/test.log
//test/core/end2end:h2_sockpair_test@ping                                FAILED in 0.1s
  /root/.cache/bazel/_bazel_root/d2dc70c3d9da3fab488ba0dcbbd35051/execroot/grpc/bazel-out/local-fastbuild/testlogs/test/core/end2end/h2_sockpair_test@ping/test.log
//test/core/end2end:h2_ssl_proxy_test@compressed_payload                 FAILED in 0.2s
  /root/.cache/bazel/_bazel_root/d2dc70c3d9da3fab488ba0dcbbd35051/execroot/grpc/bazel-out/local-fastbuild/testlogs/test/core/end2end/h2_ssl_proxy_test@compressed_payload/test.log
//test/core/end2end:h2_ssl_proxy_test@connectivity                       FAILED in 0.3s
  /root/.cache/bazel/_bazel_root/d2dc70c3d9da3fab488ba0dcbbd35051/execroot/grpc/bazel-out/local-fastbuild/testlogs/test/core/end2end/h2_ssl_proxy_test@connectivity/test.log

Executed 5037 out of 5037 tests: 5029 tests pass and 8 fail locally.

**Comment by Vizerai** [comment]
Test this please.

File: src/core/ext/census/gen/trace_context.pb.c [hunk 1]
```diff
@@ -1,6 +1,6 @@
 /*
  *
- * Copyright 2016, Google Inc.
+ * Copyright 2017, Google Inc.
```
Comment (a-veitch) [inline src/core/ext/census/gen/trace_context.pb.c]: Copyright should (I believe) be for the year the first version was submitted, not modifications. Making it a year range (2016-2017) should also be OK.

File: test/core/census/trace_context_test.c [hunk 2]
```diff
@@ -144,49 +144,48 @@ static void test_span_only() {
...
-// Test proto-buffer without is_sampled value.
-static void test_no_sample() {
+// Test proto-buffer without span_options value.
+static void test_no_span_options() {
   google_trace_TraceContext ctxt = google_trace_TraceContext_init_zero;
   read_and_validate_context_from_file(
-      &ctxt, "test/core/census/data/context_no_sample.pb", true);
-  GPR_ASSERT(ctxt.has_is_sampled == false && ctxt.is_sampled == false);
+      &ctxt, "test/core/census/data/context_no_span_options.pb", true);
+  GPR_ASSERT(ctxt.has_span_options == false && ctxt.span_options == false);
```
Comment (a-veitch) [inline test/core/census/trace_context_test.c]: Since span_options is numeric, shouldn't the last part of the condition here be "ctxt.span_options == 0" ?

Comment (Vizerai) [inline test/core/census/trace_context_test.c]: Yes, it should.... :)

**Pull request merged**